services/src/main/java/org/keycloak/authentication/DefaultAuthenticationFlow.java

```diff
@@ -308,7 +308,9 @@ void fillListsOfExecutions(Stream<AuthenticationExecutionModel> executionsToProc
                     }
                 });
 
-        if (!requiredList.isEmpty() && !alternativeList.isEmpty()) {
+        final boolean isConditionalPresent = requiredList.stream().anyMatch(AuthenticationExecutionModel::isConditional);
+
+        if (!requiredList.isEmpty() && !alternativeList.isEmpty() && !isConditionalPresent) {
```

Comment:
Is this change really needed? Should not this PR work even without the change in this file?